### PR TITLE
add and update basePath in auto-update for some libs

### PR DESCRIPTION
**Pull request by pvnr0082t**
PR to 
- add and update basePath in auto-update for some libs  
Affected 10 libs as follow:
 - bootstrap-notify
 - es6-shim
 - FileSaver.js
 - flexslider
 - jquery-visibility
 - mori
 - portal
 -  processing.js
 - punycode
 - q.js
- add test to check basePath in auto-update

cc @PeterDaveHello , cc #9192

### CODE REVIEW

File: test/valid-packages-test.js [hunk 1]
```diff
@@ -269,6 +269,18 @@ packages.map(function (pkg) {
                 pkg_name(pkg) + ": Need to add repository information in package.json");
         }
     }
+    package_vows[pname + ": There must be basePath in auto-update config"] = function(pkg) {
+        var json = parse(pkg, true);
+        if (json.npmFileMap) {
+            for (var i in json.npmFileMap) {
+                assert.ok(json.npmFileMap[i].basePath != undefined,
```
Comment (PeterDaveHello) [inline test/valid-packages-test.js]: Please make sure `basePath` is a string.

File: test/valid-packages-test.js [hunk 2]
```diff
@@ -269,6 +269,18 @@ packages.map(function (pkg) {
...
+                    pkg_name(pkg) + ": Need to add basePath in auto-update config");
+            }
+        } else if (json.autoupdate) {
+            assert.ok(json.autoupdate.basePath != undefined,
```
Comment (PeterDaveHello) [inline test/valid-packages-test.js]: Please make sure `basePath` is a string.

**Comment by pvnr0082t** [comment]
@PeterDaveHello I have fixed it. Please help me review it again, thank you.

**Comment by PeterDaveHello** [comment]
Please also update `tools/fixFormat.js`, thanks.

**Comment by PeterDaveHello** [comment]
and please also let @cdnjs/intern2 help review/test this change.

**Comment by pvnr0082t** [comment]
@cdnjs/intern2 I have updated `tools/fixFormat.js` and it can remove the leading and trailing slash in auto-update basePath.  Please help me review it thanks.

**Comment by PeterDaveHello** [comment]
@pvnr0082t it'll be better to reply on the review comment once each problem was solved.

**Comment by PeterDaveHello** [comment]
@pvnr0082t there are some conflicts need to be solved first.

**Comment by pvnr0082t** [comment]
@cdnjs/intern2 I have solved the conflict. Please help me review it, thanks.

**Comment by PeterDaveHello** [comment]
@pvnr0082t I got this error from `tools/fixFormat.js`
peter ~/temp/cdnjs $ ./tools/fixFormat.js
```
/usr/home/peter/temp/cdnjs/tools/fixFormat.js:73
          pkg.npmFileMap[i].basePath = /^\/|\/$/g[Symbol.replace](pkg.npmFileMap[i].basePath, "");
                                                                 ^

TypeError: /^\/|\/$/g[Symbol.replace] is not a function
    at /usr/home/peter/temp/cdnjs/tools/fixFormat.js:73:66
    at /usr/home/peter/temp/cdnjs/node_modules/async/dist/async.js:3047:20
    at eachOfArrayLike (/usr/home/peter/temp/cdnjs/node_modules/async/dist/async.js:1002:13)
    at eachOf (/usr/home/peter/temp/cdnjs/node_modules/async/dist/async.js:1052:9)
    at Object.eachLimit (/usr/home/peter/temp/cdnjs/node_modules/async/dist/async.js:3111:7)
    at Object.<anonymous> (/usr/home/peter/temp/cdnjs/tools/fixFormat.js:14:7)
    at Module._compile (module.js:409:26)
    at Object.Module._extensions..js (module.js:416:10)
    at Module.load (module.js:343:32)
    at Function.Module._load (module.js:300:12)
```

**Comment by PeterDaveHello** [comment]
I give jquery this filemap:
```js
[
  {
    "basePath": "/dist/",
    "files": [
      "**/*"
    ]
  }
]
```

but still pass the test.

**Comment by pvnr0082t** [comment]
@PeterDaveHello @cdnjs/intern2 For the following problems:
- Error happened because `RegExp[Symbol.replace]` only work in node versions after v6.0.0.  
 - [Fixed] I have changed to use another way to do it and it can work in node versions before v6.0.0
- Pass the test when `basePath` with the leading and trailing slash
 - The test added in this PR only check the `existence` and `string type`. Checking the leading and trailing slash test is in #9192

**Comment by PeterDaveHello** [comment]
@pvnr0082t then you should send them together in a same PR or send this one after the previous been merged.

**Comment by PeterDaveHello** [comment]
I think you mixed #9192 and #9199

**Comment by pvnr0082t** [comment]
@PeterDaveHello @cdnjs/intern2 I have remove __```update tools/fixFormat.js ```__ and commit it to #9192. 
This PR aims to fix libs without or with wrong `basePath` in auto-update and add a test to check the __```existence```__ and __```string type```__ of `basePath`. Please help me review again, thank you.

**Comment by PeterDaveHello** [comment]
Awesome :facepunch:

**Pull request merged**